modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationMetricsListener.java

```diff
@@ -326,14 +423,130 @@ private void increment() {
         }
     }
 
+    /**
+     * Metric holding information about messages network transferring time.
+     * It holds sum of time that took request to reach target node and
+     * time that took response to get back.
+     * These durations are grouped by target node id and response directType.
+     */
+    private class MsgNetworkTimeMetric {
+        /** Stores time latencies for each sent request class and each node */
+        private final Map<UUID, Map<Short, HistogramMetric>> outMetricsMap = new ConcurrentHashMap<>();
+
+        /** Is time logging enabled. */
+        private final boolean isTimeLoggingEnabled = IgniteSystemProperties.getBoolean(IGNITE_ENABLE_MESSAGES_TIME_LOGGING);
+
+        /** Metric bounds. */
+        private long[] metricBounds;
+
+        /**
+         * Reset metrics.
+         */
+        private void reset() {
+            outMetricsMap.clear();
+        }
+
+        /**
+         * Removes metrics data corresponding to {@code nodeId}
+         */
+        private void clearNodeMetrics(UUID nodeId) {
+            outMetricsMap.remove(nodeId);
+        }
+
+        /**
+         * Gets of obtains histogram metric bounds.
+         *
+         * @return Histogram metric bounds.
+         */
+        private long[] metricBounds() {
+            if (metricBounds == null)
+                metricBounds = obtainHistMetricBounds();
+
+            return metricBounds;
+        }
+
+        /**
+         * @return Metrics bounds in milliseconds.
+         */
+        private long[] obtainHistMetricBounds() {
+            String strBounds = IgniteSystemProperties.getString(IGNITE_COMM_SPI_TIME_HIST_BOUNDS);
+
+            if (strBounds == null)
+                return DEFAULT_HIST_BOUNDS;
+
+            try {
+                return Arrays.stream(strBounds.split(BOUNDS_PARAM_DELIMITER))
+                    .map(String::trim)
+                    .mapToLong(Long::parseLong)
+                    .toArray();
+            }
+            catch (Exception e) {
+                if (log != null)
```

Comment:
Missing `{`

Reply:
Not missing. log.error(...) takes 2 lines but it is still one instruction

Reply:
Braces are required in all cases, even if not required by syntax, except for the cases when next statement is **one-liner**. In this case we have two-liner.